assembler.rb

```diff
@@ -4,7 +4,8 @@
 
 class Assembler
   def initialize(input_file)
-    @parser = Parser.new(input_file)
+    @input_file = input_file
+    @parser = Parser.new(@input_file)
```

Comment:
There’s no longer much value in creating the parser in the constructor, or in keeping it in an instance variable at all, because we know we’re going to need two of them anyway. We might as well just create them both inside `#assemble` and keep them in a local variable. That would make the code for `#assemble` more regular and self-contained, instead of this slightly surprising situation where the first parser is created in the constructor and the second is created inside `#assemble` for (I think) no intentional reason.

Reply:
Good idea, if I were looking at the code for the first time, I'd be confused to why there's one parser created in the initialize but another in `#assemble`. It'd be clearer to have them both in the same method.